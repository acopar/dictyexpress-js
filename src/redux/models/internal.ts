import {
    DataDifferentialExpression,
    GOEnrichmentJson,
    GOEnrichmentNode,
    Relation,
    Storage,
} from '@genialis/resolwe/dist/api/types/rest';
import { Feature } from '@genialis/resolwe/dist/api/types/modules';
import { VariantType, SnackbarKey } from 'notistack';
import { Renderers } from 'vega';
import { LayoutBreakpoint } from 'components/genexpress/common/constants';

export type Gene = Pick<
    Feature,
    'feature_id' | 'aliases' | 'name' | 'full_name' | 'description' | 'source' | 'species'
>;

export type RelationsById = {
    [_: number]: Relation;
};

export type GenesById = {
    [_: string]: Gene;
};

export type BasketInfo = {
    id: string;
    source: string;
    species?: string;
    type: 'gene';
};

export type GeneSet = {
    dateTime: Date;
    genesNames: string[];
};

export type GeneExpression = {
    [geneId: string]: number;
};

export type SamplesExpressionsById = {
    [sampleId: number]: GeneExpression;
};

export type VolcanoPoint = {
    geneId: string; // Feature (gene) id
    logFcValue: number;
    logProbValue: number;
    logProbFiniteValue: number;
    probValue: number;
};

export type Thresholds = {
    pValue: number;
    pValueLog: number;
    fc: number;
    fcLog: number;
};

export type DifferentialExpression = {
    logfc_threshold: number;
    prob_field: string;
    prob_threshold: number;
    up_regulated: number;
    down_regulated: number;
} & DataDifferentialExpression &
    Partial<Storage['json']>;

export type DifferentialExpressionsById = {
    [differentialExpressionId: number]: DifferentialExpression;
};

export type AspectValue = 'BP' | 'CC' | 'MF';
export type Aspect = {
    value: AspectValue;
    label: string;
};

export type EnhancedGOEnrichmentJson = {
    tree: {
        [aspectSlug: string]: GOEnrichmentRow[];
    };
} & Omit<GOEnrichmentJson, 'tree'>;

export type GOEnrichmentRow = {
    path: string[];
    children?: GOEnrichmentRow[];
    score_percentage: number;
    gene_associations: string[];
} & Omit<GOEnrichmentNode, 'children'>;

// If key is undefined, it will be generated in the reducer.
export type SnackbarNotificationContent = {
    key?: SnackbarKey;
    message: string;
    variant?: VariantType;
    action?: (key: number) => void;
};

// TODO, is this the correct way of going about making only one property required?
export type SnackbarNotification = {
    key: SnackbarKey;
} & Pick<SnackbarNotificationContent, 'message' | 'variant' | 'action'>;

export type SnackbarNotifications = SnackbarNotification[];

export type DictyRenderers = Exclude<Renderers, 'none'>;

export type BreakpointsCols = { [key in keyof typeof LayoutBreakpoint]: number };
