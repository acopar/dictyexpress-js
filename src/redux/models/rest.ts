export type BasketAddSamplesResponse = {
    id: string;
    modified: string;
    ignored: number[];
    permitted_organisms: string[];
    permitted_sources: string[];
    conflict_organisms: string[];
    conflict_sources: string[];
};
