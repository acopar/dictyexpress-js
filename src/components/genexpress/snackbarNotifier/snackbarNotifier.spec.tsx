import React from 'react';
import { customRender } from 'tests/test-utils';
import { mockStore, testState } from 'tests/mock';
import { SnackbarNotifications } from 'redux/models/internal';
import SnackbarNotifier from './snackbarNotifier';

const initialTestState = testState();
const testNotifications: SnackbarNotifications = [
    { key: '1', message: 'Test message', variant: 'default', action: (): void => {} },
    { key: '2', message: 'Test message 2', variant: 'error', action: (): void => {} },
    { key: '3', message: 'Test message 3', variant: 'success', action: (): void => {} },
    { key: '4', message: 'Test message 4', variant: 'info', action: (): void => {} },
    { key: '5', message: 'Test message 5', variant: 'warning', action: (): void => {} },
];

initialTestState.notifications.notifications = testNotifications;

const mockedStore = mockStore(initialTestState);

describe('snackbarNotifier', () => {
    it('should render as snapshot with notifications in all different variants', () => {
        const { asFragment } = customRender(<SnackbarNotifier />, {
            mockedStore,
        });

        expect(asFragment()).toMatchSnapshot();
    });
});
