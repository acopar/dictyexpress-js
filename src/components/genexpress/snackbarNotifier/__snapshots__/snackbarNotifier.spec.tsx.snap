// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snackbarNotifier should render as snapshot with notifications in all different variants 1`] = `
<DocumentFragment>
  <div
    class="SnackbarContainer-bottom SnackbarContainer-left SnackbarContainer-root css-uwcd5u"
  >
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="SnackbarItem-wrappedRoot css-1taihvu"
          >
            <div
              aria-describedby="notistack-snackbar"
              class="SnackbarContent-root SnackbarItem-contentRoot css-hped4j"
              role="alert"
              style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
            >
              <div
                class="SnackbarItem-message"
                id="notistack-snackbar"
              >
                Test message
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="SnackbarItem-wrappedRoot css-1taihvu"
          >
            <div
              aria-describedby="notistack-snackbar"
              class="SnackbarContent-root SnackbarItem-contentRoot SnackbarItem-variantError SnackbarItem-lessPadding css-hped4j"
              role="alert"
              style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
            >
              <div
                class="SnackbarItem-message"
                id="notistack-snackbar"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  focusable="false"
                  style="font-size: 20px; margin-inline-end: 8px;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,2C17.53,2 22,6.47 22,12C22,17.53 17.53,22 12,22C6.47,22 2,17.53 2,12C2,
        6.47 6.47,2 12,2M15.59,7L12,10.59L8.41,7L7,8.41L10.59,12L7,15.59L8.41,17L12,
        13.41L15.59,17L17,15.59L13.41,12L17,8.41L15.59,7Z"
                  />
                </svg>
                Test message 2
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="SnackbarItem-wrappedRoot css-1taihvu"
          >
            <div
              aria-describedby="notistack-snackbar"
              class="SnackbarContent-root SnackbarItem-contentRoot SnackbarItem-variantSuccess SnackbarItem-lessPadding css-hped4j"
              role="alert"
              style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
            >
              <div
                class="SnackbarItem-message"
                id="notistack-snackbar"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  focusable="false"
                  style="font-size: 20px; margin-inline-end: 8px;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M10 17L5 12L6.41
        10.59L10 14.17L17.59 6.58L19 8L10 17Z"
                  />
                </svg>
                Test message 3
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
